### PR TITLE
Adding automatic testing of the applications

### PR DESCRIPTION
**Pull request by sebglane**
This branch includes a merge of #67 because otherwise the tests are not executable.

### CODE REVIEW

**Comment by sebglane** [comment]
Merge after #123

File: applications/MIT.cc [hunk 1]
```diff
@@ -201,31 +199,53 @@ flag_local_refinement(true)
...
+        else
+          Assert(false, ExcInternalError());
+      }
+      else if (std::abs(center[1]) < tol)
+        face->set_boundary_id(bottom_bndry_id);
+      else if (std::abs(center[1] - 8.0) < tol)
+        face->set_boundary_id(top_bndry_id);
+      else
+        Assert(false, ExcInternalError());
+    }
```
Comment (j507) [inline applications/MIT.cc:235]: I know this mesh is a quick fix but I am not really satisfied with it. Its improvement in order to reflect the tailored structured mesh should be added to the to-do list.

File: source/benchmark_data.cc [hunk 2]
```diff
@@ -34,10 +34,10 @@ DFGBechmarkRequest<dim>::DFGBechmarkRequest
...
 :
 Re(reynolds_number),
-front_evaluation_point(0.15 / reference_length,
-                       0.20 / reference_length),
-rear_evaluation_point(0.25 / reference_length,
-                      0.20 / reference_length),
+front_evaluation_point(1.5 / reference_length,
+                       2.0 / reference_length),
+rear_evaluation_point(2.5 / reference_length,
+                      2.0 / reference_length),
```
Comment (j507) [inline source/benchmark_data.cc:40]: This is wrong. More specifically, the default value of `reference_length` is wrong. It should be `0.1`. The other values remain unchanged.

**Review by j507 — CHANGES_REQUESTED** [review]
I only skimmed through the `yml` file as I am not familiar with the commands. I would think that since the meshes of `DFG.cc` and `MIT.cc` were changed, a re run of the benchmark is in order. The mesh for `MIT.cc` is over simplified and needs works but that can be postponed. I assume the mesh for `DFG.cc` is just a copy paste of the `dealii` method. 

The only important change before continuing is the correction of the `reference_length` in the DFG benchmark data. Once that is corrected, feel free to merge.

File: CMakeLists.txt [hunk 3]
```diff
@@ -4,7 +4,7 @@
 MESSAGE("====================================================")
 MESSAGE("=============== Configuring RotatingMHD ============")
 MESSAGE("========= An adaptive parallel flow solver =========")
-MESSAGE("========= for the rotating magnetohydrodynamics =====")
+MESSAGE("========= for the rotating magnetohydrodynamics ====")
```
[the remark this answers is not in the document]
Comment (sebglane) [reply CMakeLists.txt]: Fixed in faac444.

File: applications/DFG.cc [hunk 4]
```diff
@@ -352,35 +355,259 @@ pressure_initial_condition()
...
+
+  for (unsigned int i=0;
+       i<prm.spatial_discretization_parameters.n_initial_boundary_refinements;
+       ++i)
+  {
+    for (auto &cell: this->triangulation.active_cell_iterators())
+      if (cell->at_boundary() && cell->is_locally_owned())
+        cell->set_refine_flag();
+    this->triangulation.execute_coarsening_and_refinement();
+  }
```
[the remark this answers is not in the document]
Comment (sebglane) [reply applications/DFG.cc]: I adjusted the mesh again in 358eb92. I did not re-run the complete benchmark. Could you do that?

File: applications/MIT.cc (hunk 1 above)
Comment (sebglane) [reply applications/MIT.cc]: The mesh was optimized in a3b6a67. Let me know your opinion on the current version.

File: source/benchmark_data.cc (hunk 2 above)
Comment (sebglane) [reply source/benchmark_data.cc]: I don't understand your comment. The cylinder has a diameter of 1 which means that we scale the spatial coordinates specified by the DFG benchmark by a factor of 10. Therefore the original coordinates of the evaluation points (0.15, 0.2) and (0.25, 0.2) become (1.5, 2.0) and (2.5, 2.0). I removed the variable `reference_length` in da2d96e.

**Comment by sebglane** [comment]
Could you comment on my changes and sign this off? For running the benchmarks again I need the specific input files.

**Comment by j507** [comment]
Code compiles on my end but I get segmentation faults when running the applications. They run on the cluster so I will have to figure out whats wrong on my side.

**Comment by j507** [comment]
Fixed the problem on my end. Could you merge master in this branch before I review it? There are over 52 changed files as of now.

**Comment by sebglane** [comment]
> Fixed the problem on my end. Could you merge master in this branch before I review it? There are over 52 changed files as of now.

I merged master. This changed the number of modified files to 49.

**Review by j507 — APPROVED** [review]

**Pull request merged**